src/gluonts/model/naive_2/_predictor.py

```diff
@@ -0,0 +1,167 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+# Standard library imports
+from typing import Iterator, Optional
+
+# Third-party imports
+import numpy as np
+import pandas as pd
+
+# First-party imports
+from gluonts.core.component import validated
+from gluonts.dataset.common import Dataset, DataEntry
+from gluonts.model.forecast import SampleForecast, Forecast
+from gluonts.model.predictor import RepresentablePredictor
+from gluonts.evaluation import get_seasonality
+import statsmodels.api as sm
+
+
+def seasonality_test(past_ts_data: np.array, season_length: int):
+    """
+    Test the time-series for seasonal patterns by performing a 90% auto-correlation test:
+
+    As described here: https://www.m4.unic.ac.cy/wp-content/uploads/2018/03/M4-Competitors-Guide.pdf
+    Code based on: https://github.com/Mcompetitions/M4-methods/blob/master/Benchmarks%20and%20Evaluation.R
+    """
+    critical_z_score = 1.645  # corresponds to 90% confidence interval
+    if len(past_ts_data) < 3 * season_length:
+        is_seasonal = False
+    else:
+        # calculate auto-correlation for lags up to season_length
+        auto_correlations = sm.tsa.stattools.acf(
+            past_ts_data, fft=False, nlags=season_length
+        )
+        auto_correlations[1:] = 2 * auto_correlations[1:] ** 2
+        limit = (
+            critical_z_score
+            / np.sqrt(len(past_ts_data))
+            * np.sqrt(np.cumsum(auto_correlations))
+        )
+        is_seasonal = (
+            abs(auto_correlations[season_length]) > limit[season_length]
+        )
+
+    return is_seasonal
+
+
+def naive_02(
```

Comment:
A minor thing: the model subpackage is called `naive_2`, you can call this function too `naive_2`, as well as other occurrences in the tests.

Reply:
Actually, not so minor

Reply:
Also in file names in the tests: it's better to be consistent with `naive_2`, consistency is good